### PR TITLE
Increase PPU Accuracy and Performance

### PR DESCRIPTION
**Pull request by zer0x64**
This PR should make the PPU cycle accurate (assuming there are no mistakes)

~~I still have a small bug to fix before it's ready to merge. Sometimes, sprites that falls off the bottom of the screen are shown on the top 1 pixel of the screen.~~ ~~I've identified the bug, now I'm looking if it's intended and supposed to be hidden by overscan, if not, why it doesn't happen on the real nes to do an accurate fix~~

~~Frame time is slighty different when rendering is disabled, so that's something I need to address without merging.~~ Fixed

This fixes a lot of bugs, especially graphical one, and a lot of game seems to finally work perfectly after this.

Closes https://github.com/zer0x64/nestadia/issues/43

### CODE REVIEW

File: nestadia/src/ppu/mod.rs [hunk 1]
```diff
@@ -47,37 +68,47 @@ impl Ppu {
...
-        self.addr_reg = registers::VramAddr::default();
-        self.cycle_count = 0;
-        self.scanline = 0;
-        self.frame = [0u8; FRAME_WIDTH * FRAME_HEIGHT];
-        self.vblank_nmi_set = false;
-        self.sprite_zero_in_secondary_oam = false;
+        *self = Self {
+            scanline: -1,
+            ..Self::default()
+        };
```
Comment (CBenoit) [inline nestadia/src/ppu/mod.rs:111]: Default value already sets scanline to `-1`
```suggestion
        *self = Self::default();
```

File: nestadia/src/ppu/mod.rs [hunk 2]
```diff
@@ -273,59 +326,123 @@ impl Ppu {
...
             }
         };
 
-        self.render_pixel(bus);
-    }
+        self.render_pixel();
 
-    fn render_pixel(&mut self, bus: &mut PpuBus) {
-        use core::convert::TryFrom;
+        // This condition is there to ensure that the VRAM address does not get updated if rendering is turned off
```
Comment (CBenoit) [inline nestadia/src/ppu/mod.rs:399]: But it could get updated just above (line 393) though?

**Review by CBenoit — APPROVED** [review]
Now it looks like a real emulator!

File: nestadia/src/ppu/mod.rs (hunk 1 above)
Comment (zer0x64) [reply nestadia/src/ppu/mod.rs:111]: I didn't see there was a default implementation!

File: nestadia/src/ppu/mod.rs (hunk 2 above)
Comment (zer0x64) [reply nestadia/src/ppu/mod.rs:399]: That was a partial refactor! The RENDER_PIXEL || RENDER_BACKGROUND condition means rendering is enabled, and I've extracted it in a function but forgot to change it right above line 393. I'm fixing it now!

**Comment by zer0x64** [comment]
Ready to merge if you're fine with it!

**Comment by CBenoit** [comment]
Sounds good to me!

**Pull request merged**